ray_lightning/ray_ddp.py

```diff
@@ -459,9 +513,14 @@ def node_rank(self) -> int:
     @property
     def root_device(self):
         if self.use_gpu and torch.cuda.is_available():
-            # Ray already sets CUDA_VISIBLE_DEVICES for each process.
-            # So the device is the 0th index in CUDA_VISIBLE_DEVICES
-            return torch.device("cuda", 0)
+            if self._is_remote:
+                # Adjust for if there are multiple GPUs per worker.
+                device_id = self.local_rank + (self.num_gpus_per_worker - 1)
```

Comment:
What's the intended logic here? 

If `num_gpus_per_worker=n` this would give `[n-1, n, n+1, ...]`?

Reply:
yeah idk what I was thinking here 😅. Fixed with correct math and added a test for this.